src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -2632,10 +3118,10 @@ public class C1
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
 
-        [Fact]
-        public void InStruct()
+        [Theory, CombinatorialData]
+        public void InStruct(bool structTreeFirst)
```

Comment:
>InStruct [](http://example.com/codeflow?start=20&length=8)

We also need to test for possible cycles between two (or more) structures:
- cyclic declaration (structures declare fields of each other)
- declarations are not cyclic, but properties with `field` keyword create an instance of the other structure
- etc.

Reply:
This is a very interesting scenario. I'll add tests to confirm whether we get the correct behavior.

Reply:
Fixed and added tests. They currently cause extra binding :(
I'll see if I can resolve that.

Reply:
> Fixed and added tests. 

Do we have a test for
"declarations are not cyclic, but properties with field keyword create an instance of the other structure"?

Reply:
> "declarations are not cyclic, but properties with field keyword create an instance of the other structure"?

That seems applicable to static properties? Is that what you have in mind? If so, there is `InStruct_NoCycleBecauseStatic`

Reply:
>> "declarations are not cyclic, but properties with field keyword create an instance of the other structure"?

> That seems applicable to static properties? Is that what you have in mind? If so, there is InStruct_NoCycleBecauseStatic

No, the scenario I have in mind is not about static properties. I am thinking about scenario where there is no cycle due to property types (declarations are not cyclic), but properties with field keyword creating an instance of the other structure (accessors can do many different things, not just reading/writing `field`). Not returning the instance, but creating it. Perhaps in a manner that would trigger some interesting data/flow analysis. For example:
```
struct S1
{
    int P1 
    {
        get
        {
              // use `field` keyword
              // Create an instance of S2 in some interesting way.
        }
    }
}   

struct S2
{
    int P2 
    {
        get
        {
              // use `field` keyword
              // Create an instance of S1 in some interesting way.
        }
    }
}   
```

Reply:
@AlekseyTs Something like?

```csharp
struct S1
{
    int P1 
    {
        get
        {
              _ = field;
              var x = new S2();
              return field;
        }
    }
}   

struct S2
{
    int P2 
    {
        get
        {
              _ = field;
              var x = new S1();
              return field;
        }
    }
}
```

I don't see why the above scenario would be interesting.
Is it what you have in mind?

> Create an instance of S1 in **some interesting way**.

Probably I can't figure the "some interesting way"

Reply:
> Something like?

> Probably I can't figure the "some interesting way"

Probably an interesting scenario will be a scenario when no matter which structure type is bound first (forcing binding for specific syntax tree), we will get an extra accessor binding for the other struct due to flow analysis.

```new``` fully assigns the struct, I am not sure if that is "interesting" enough. Perhaps use of an unassigned local could be more "interesting". This requires some experimentation.

Reply:
@AlekseyTs

> when no matter which structure type is bound first (forcing binding for specific syntax tree), we will get an extra accessor binding for the other struct due to flow analysis.

Are we looking for a test where `NumberOfPerformedAccessorBinding` is 1 when the structs aren't circular?

If that's the case, then the code I mentioned above does it. The following test passes:

```csharp
        [Fact]
        public void InStruct_NoCycle()
        {
            var source1 = @"
struct S1
{
    public S1() { }

    public int P
    {
        get
        {
            _ = field;
            var x = new S2();
            return field;
        }
    }
}";

            var source2 = @"
struct S2
{
    public S2() { }

    public int P
    {
        get
        {
            _ = field;
            var x = new S1();
            return field;
        }
    }
}";
            var comp = CreateCompilation(new[] { source1, source2 });
            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
            comp.TestOnlyCompilationData = accessorBindingData;
            comp.GetDiagnosticsForSyntaxTree(CompilationStage.Compile, comp.SyntaxTrees[0], filterSpanWithinTree: null, includeEarlierStages: true).Verify();
            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding);
            comp.GetDiagnosticsForSyntaxTree(CompilationStage.Compile, comp.SyntaxTrees[1], filterSpanWithinTree: null, includeEarlierStages: true).Verify();
            Assert.Equal(1, accessorBindingData.NumberOfPerformedAccessorBinding);
        }
```

Reply:
It seems it's caused by flow analysis. Here is the callstack:

![image](https://user-images.githubusercontent.com/31348972/167922245-2a96ed59-19be-465b-95eb-e9bdbfca32c8.png)